projects/js-packages/charts/src/components/leaderboard-chart/leaderboard-chart.tsx

```diff
@@ -133,25 +142,65 @@ const defaultDeltaFormatter = ( value: number ): string => {
 	} );
 };
 
+const ProgressBarWithOverlayLabel = ( { entry }: { entry: LeaderboardEntry } ) => (
+	<div className={ styles.progressContainerWithOverlayLabel }>
+		{ typeof entry.label === 'string' ? (
+			<Text className={ styles.progressBarLabel }>{ entry.label }</Text>
+		) : (
+			entry.label
+		) }
+
+		<div className={ styles.progressBar } style={ { width: entry.currentShare + '%' } }></div>
+	</div>
+);
+
+const ProgressBarWithLabel = ( {
+	entry,
+	withComparison,
+}: {
+	entry: LeaderboardEntry;
+	withComparison?: boolean;
+} ) => (
+	<>
+		{ typeof entry.label === 'string' ? <Text>{ entry.label }</Text> : entry.label }
+
+		<div className={ styles.progressContainer }>
+			<ProgressBar
```

Comment:
It seems ProgressBar wouldn't satisfy the needs here, so we should probably just use our custom implementation in ProgressBarWithOverlayLabel.